feat: Added unit tests for webhook-service and improved error messages

odubajDT opened the pull request:
Signed-off-by: odubajDT <ondrej.dubaj@dynatrace.com>

<!-- Please use this template for your pull request. -->
<!-- Please use the sections that you need and delete other sections -->

## This PR
- adds unit tests for webhook-config and improved error messages

### Related Issues
<!-- add here the GitHub issue that this PR resolves if applicable -->

Fixes #5692

bacherfl commented on webhook-service/handler/handler.go:
```diff
@@ -84,7 +85,7 @@ func (th *TaskHandler) Execute(keptnHandler sdk.IKeptn, event sdk.KeptnEvent) (i
 		err = keptnHandler.SendFinishedEvent(event, result)
 		if err != nil {
-			return nil, sdkError(fmt.Sprintf("could not send finished event: %s", err.Error()), err)
+			return nil, sdkError(fmt.Sprintf("could not send finished event: request '%s' %s", webhook.Requests[0], err.Error()), err)
```
Can it happen that `webhook.Requests` is empty at this point? Also, is the first of the webhook requests actually tied to this particular error (sending the finished event failed)?

odubajDT replied:
Actually it shouldn't happen, we are now checking that webhook.Requests are not empty, see this PR below

bacherfl replied:
ok thank you for clarifying :) But still, adding the first of the requests to this error message might be misleading, as the error in this case does not see, to be tied to that particular request

bacherfl commented on webhook-service/handler/handler.go:
```diff
@@ -170,13 +171,13 @@ func (th *TaskHandler) onStartedWebhookExecution(keptnHandler sdk.IKeptn, event
 		// the webhook service will then send a correlating .finished event with the aggregated response payloads
 		if err := keptnHandler.SendStartedEvent(event); err != nil {
-			return sdkError(fmt.Sprintf("could not send .started event: %s", err.Error()), err)
+			return sdkError(fmt.Sprintf("could not send .started event: request '%s': %s", webhook.Requests[0], err.Error()), err)
```
Can it happen that `webhook.Requests` is empty at this point? Also, is the first of the webhook requests actually tied to this particular error (sending the started event failed)?

odubajDT replied:
Actually it shouldn't happen, we are now checking that webhook.Requests are not empty, see this PR below

odubajDT commented on webhook-service/lib/webhook_config.go:
```diff
@@ -44,5 +48,24 @@ func DecodeWebHookConfigYAML(webhookConfigYaml []byte) (*WebHookConfig, error) {
+			return nil, errors.New("Webhook configuration invalid: missing 'webhooks[].SubscriptionID' part")
+		}
+
+		if len(webhook.Requests) == 0 {
```
@bacherfl in this place

bacherfl commented on webhook-service/lib/template_engine_test.go:
```diff
@@ -42,6 +43,46 @@ func TestTemplateEngine_ParseTemplate(t1 *testing.T) {
 			want:    "",
 			wantErr: true,
 		},
+		{
```
Is there a way to get any information on what field of a template string was causing the error? I.e. like described in the issue (https://github.com/keptn/keptn/issues/5692)

odubajDT replied:
will look at it :)

thisthat replied:
Maybe this could help here: https://pkg.go.dev/text/template#Template.Option

odubajDT replied:
will add, thanks!

odubajDT replied:
we already have this Option() function used in ParseTemplate(), see https://github.com/keptn/keptn/blob/master/webhook-service/lib/template_engine.go#L16

odubajDT commented:
@bacherfl please re-review? I removed the request[0] as they really do not make sense there and added the required errMsg to the unit tests, hope now it will be good :)
